txs/conservative_state.go

```diff
@@ -146,67 +155,98 @@ func (cs *ConservativeState) RevertState(revertTo types.LayerID) (types.Hash32,
 }
 
 // ApplyLayer applies the transactions specified by the ids to the state.
-func (cs *ConservativeState) ApplyLayer(toApply *types.Block) ([]*types.Transaction, error) {
+func (cs *ConservativeState) ApplyLayer(toApply *types.Block) ([]types.TransactionID, error) {
 	logger := cs.logger.WithFields(toApply.LayerIndex, toApply.ID())
 	logger.Info("applying layer to conservative state")
 
 	if err := cs.cache.CheckApplyOrder(toApply.LayerIndex); err != nil {
 		return nil, err
 	}
 
-	txs, err := cs.getTXsToApply(toApply)
+	txs, raw, err := cs.getTXsToApply(toApply)
 	if err != nil {
 		return nil, err
 	}
 
-	failedTxs, err := cs.vmState.ApplyLayer(toApply.LayerIndex, txs, toApply.Rewards)
+	skipped, err := cs.vmState.Apply(toApply.LayerIndex, raw, toApply.Rewards)
```

Comment:
couldn't you pass []Transaction to vmState.Apply()? since a Transaction is implicitly a RawTx, is it possible for vmState.Apply() to take []*Transaction or []*RawTx. that way you can keep the return types getTXsToApply unchanged.

Reply:
vm doesn't need anything beyond raw tx, by providing txs with headers someone can decide to reuse headers in the vm

Reply:
i was thinking doing
`skipped, err := cs.vmState.Apply(toApply.LayerIndex, txs, toApply.Rewards)`
(assuming txs is []Transaction)
and keep vmState.Apply() to take []RawTx. 

does that work with golang 1.8, since a Transaction is a RawTx also?

Reply:
it definitely won't work with slice. not sure if it will work with a single object